api/blast_ui.api.inc

```diff
@@ -1,12 +1,96 @@
 <?php
-
 /**
  * @file
  * Contains more generally applicable functions as well as some meant to help developers
  * Plug-in to the BLAST UI functionality
  */
 
 /**
+ * This program is used for services module specifically.
+ * Run BLAST (should be called from the command-line)
+ *
+ * @param $program
+ *   Which BLAST program to run (ie: 'blastn', 'tblastn', tblastx', 'blastp','blastx')
+ * @param $query
+ *   The full path and filename of the query FASTA file
+ * @param $database
+ *   The full path and filename prefix (excluding .nhr, .nin, .nsq, etc.)
+ * @param $output_filestub
+ *   The filename (not including path) to give the results. Should not include file type suffix
+ * @param $options
+ *   An array of additional option where the key is the name of the option used by
+ *   BLAST (ie: 'num_alignments') and the value is relates to this particular
+ *   BLAST job (ie: 250)
+ * @param $format 
+ *   Which specifies Json or asn format of file or program
+ * @param  $format_type 
+ * which specify the number of format used by BLAST e.g json has 11 and json as 12. 
+ *
+ * Author: Prateek Gupta
+ * Created: 2016
+ */
+
+function run_BLAST_tripal_job_services($program, $query, $database, $output_filestub, $options, $format, $format_type,  $job_id = NULL) {
+  $output_file = file_directory_temp() .  DIRECTORY_SEPARATOR . $output_filestub . '.blast.' . $format_type;
+  $output_dir = variable_get('file_public_path', conf_path() . '/files') 
+              . DIRECTORY_SEPARATOR . 'tripal' . DIRECTORY_SEPARATOR . 'tripal_blast';
+
+  print "\nExecuting $program\n\n";
+  print "Query: $query\n";
+  print "Database: $database\n";
+  print "Results File: $output_file\n";
+
+  print "Options:\n";
+
+  // Allow administrators to use an absolute path for these commands.
+  // Defaults to using $PATH.
+  $blast_path    = variable_get('blast_path', '');
+  $blast_threads = variable_get('blast_threads', '');
+  
+  // Strip the extension off the BLAST target
+  $database = preg_replace("/(.*)\.[pn]\w\w/", '$1', $database);
```

Comment:
We need to be consistent with how we are handling this. Here you have the old code (very buggy) and in the original run blast function you've commented it out entirely. I suggest we stick with my most recent fix of changing the regex to ensure the "file extension" captured is at the end and only 3 characters for now since. If you feel strongly that this issue should be dealt with immediately, I don't want to comment it out until validation is added to the admin form to node form to ensure that admin don't enter the extension in the first place.